src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -91,42 +104,43 @@ extends AbstractMap[Key, Value]
     table = new Array[Entry](newSize)
     mask = newSize - 1
     oldTable.foreach( entry =>
-      if (entry != null && entry.value != None) addEntry(entry))
+      if (entry != null && entry.value != None)
```

Comment:
Wouldn't it be a good idea to now check if the `entry.key` is `null`, and skip entries whose keys are `null`. 

Alternatively, if it makes sense to reuse those objects, one could just insert them anywhere.